### PR TITLE
Fix bug with greeting [BOTKIT]

### PR DESCRIPTION
**Pull request by nikitasakau**
Fix https://github.com/datarockets/mona/issues/53 https://github.com/datarockets/mona/issues/31
image
- Updated `botbuilder-adapter-slack`, `botkit`, `botkit-mock` in order to use middlewares in specs (Only the last version `botkit-mock` supports them)
- Changed some styles in specs as in examples of botkit-mock
- Fixed bug with greeting as we fixed it in master
- Created spec that cover this bug

### CODE REVIEW

File: specs/greeting_spec.js [hunk 1]
```diff
@@ -1,40 +1,52 @@
...
-    this.controller = Botmock({})
-    this.bot = this.controller.spawn({ type: 'slack' })
+    this.controller = getBasicController()
     greetingController(this.controller)
   })
 
-  it(
-    'Should return any greeting if user types `hi`',
-    () => this.bot.usersInput([{
+  it('Should return any greeting if user types `hi`', async () => {
```
Comment (AleksSenkou) [inline specs/greeting_spec.js]: without `should` 😉 
```
it 'returns '
```

http://www.betterspecs.org/
**Don't use should** section

File: specs/greeting_spec.js [hunk 2]
```diff
@@ -1,40 +1,52 @@
...
-    () => this.bot.usersInput([{
+  it('Shouldn\'t return any greeting if user types a word that includes greeting word', async () => {
+    await this.controller.usersInput([{
+      type: 'message',
+      channel: 'channel',
       messages: [{
-        text: '```hi```', isAssertion: true,
+        text: 'they', isAssertion: true,
       }],
     }]).then(message => assert.deepEqual(message, {}))
```
Comment (AleksSenkou) [inline specs/greeting_spec.js:50]: thoughts about moving this method to a helper method for tests? It's duplicated lot of times

**Review by AleksSenkou — COMMENTED** [review]
Nice to see tests rebuilt 👍

File: specs/greeting_spec.js (hunk 2 above)
Comment (nikitasakau) [reply specs/greeting_spec.js:50]: @AleksSenkou Do you mean `message => assert.deepEqual(message, {})` only or the whole `usersInput`?

**Comment by npupko** [comment]
@nikitasakov I'm not sure it related to, but should warnings be removed in this MR? 
I.e. 
```
[WARN]  WebClient:1 http request failed
```

Or does it not affect anything? Everyone usually has a different attitude towards the warnings.

**Review by npupko — APPROVED** [review]

**Comment by npupko** [comment]
I need this branch to be merged to finish write tests for [order water feature](https://github.com/datarockets/mona/issues/24), dudes. 
Will cherry-pick it locally for now

File: specs/greeting_spec.js (hunk 2 above)
Comment (AleksSenkou) [reply specs/greeting_spec.js:50]: @nikitasakov I'm thinking about:
```js
await userInputs('they').then(message => assert.deepEqual(message, {}))
```

thoughts?

Comment (lazycoder9) [reply specs/greeting_spec.js:50]: Why do you use await on promise?

Comment (AleksSenkou) [reply specs/greeting_spec.js:50]: or:

```js
await userInputs('they').then(noAnswerFromBotExpected)
```

**Comment by nikitasakau** [comment]
@npupko Do you mean that we can merge it for now, but create separate issue for refactoring?

File: specs/greeting_spec.js (hunk 2 above)
Comment (nikitasakau) [reply specs/greeting_spec.js:50]: @lazycoder9 I did as in docs - https://github.com/gratifyguy/botkit-mock. Or it is not the same?

**Comment by npupko** [comment]
@nikitasakov Yes, it will be awesome. How much time do you need to complete this issue?

**Comment by AleksSenkou** [comment]
@nikitasakov what about this comment? https://github.com/datarockets/mona/pull/59#discussion_r352631339

**Comment by nikitasakau** [comment]
@AleksSenkou We decided to refactor it in separate issue. Going to continue discussion there. https://github.com/datarockets/mona/issues/61

**Pull request merged**